libraries/AP_HAL_Linux/VideoIn.cpp

```diff
@@ -235,6 +255,48 @@ void VideoIn::prepare_capture()
     }
 }
 
+void VideoIn::crop_image_grey(uint8_t *buffer, uint32_t buffer_size, uint8_t *new_buffer, uint32_t width,
+    uint32_t height, uint32_t cropLeft, uint32_t cropRight, uint32_t cropTop, uint32_t cropBottom)
+{
+    uint32_t i, new_buffer_position = 0, columnCount = 0, lineCount = 0;
+
+    for (i = 0; i < buffer_size; i++) {
+        if (columnCount == width) {
+            columnCount = 0;
+            lineCount++;
+        }
+
+        if (columnCount < cropLeft || columnCount >= (width - cropRight)) {
+            columnCount++;
+            continue;
+        }
+
+        if (lineCount < cropTop || lineCount >= (height - cropBottom)) {
+            columnCount++;
+            continue;
+        }
+
+        columnCount++;
+
+        new_buffer[new_buffer_position] = buffer[i];
+
+        new_buffer_position++;
+    }
+}
```

Comment:
Ok, I get it, the notations are different. Usually, i.e in v4l2, crop right is the last column and crop left, the first, and crop top the first line and crop bottom the last one. I hadn't noticed you were using a different notation.